File: XWeather/iOS/Extensions/ViewExtensions.cs
using System;

using CoreAnimation;
using Foundation;
using UIKit;

namespace XWeather.iOS
{
	public static class ViewExtensions
	{

		public static void SetTransparencyMask (this UIView view, nfloat top, nfloat bottom)
		{
			if (top > view.Frame.Height || bottom > view.Frame.Height) {

				view.Layer.Mask = view.visibilityMaskWithLocation (1);

			} else if (top >= 0 && top <= view.Frame.Height) {

				var location = top / view.Frame.Height;

				view.Layer.Mask = view.visibilityMaskWithLocation (location);

			} else if (bottom >= 0 && bottom <= view.Frame.Height) {

				var location = NMath.Abs (1 - (bottom / view.Frame.Height));

				view.Layer.Mask = view.visibilityMaskWithLocation (location, true);

			} else {

				view.Layer.Mask = null;
			}

			view.Layer.MasksToBounds = true;
		}


		static CAGradientLayer visibilityMaskWithLocation (this UIView view, nfloat location, bool reverse = false)
		{
			var mask = new CAGradientLayer ();

			mask.Frame = view.Bounds;

			if (reverse) {

				mask.Colors = new [] { UIColor.FromWhiteAlpha (1, 1).CGColor, UIColor.FromWhiteAlpha (1, 0).CGColor };

			} else {

				mask.Colors = new [] { UIColor.FromWhiteAlpha (1, 0).CGColor, UIColor.FromWhiteAlpha (1, 1).CGColor };
			}

			mask.Locations = new [] { NSNumber.FromNFloat (location), NSNumber.FromNFloat (location) };

			return mask;
		}
	}
}

File: Swift/XWeather/XWeather/Extensions/ViewExtensions.swift
//
//  ViewExtensions.swift
//  XWeather
//
//  Created by Colby Williams on 9/13/16.
//  Copyright © 2016 Colby Williams. All rights reserved.
//

import CoreGraphics
import Foundation
import UIKit

extension UIView {
	
	func setTransparencyMask(top: CGFloat, bottom: CGFloat) {
		
		if top > self.frame.height || bottom > self.frame.height {
			
			self.layer.mask = self.visibilityMaskWithLocation(1)
		
		} else if top >= 0 && top <= self.frame.height {
			
			let location = top / self.frame.height
			
			self.layer.mask = self.visibilityMaskWithLocation(location)
			
		} else if bottom >= 0 && bottom <= self.frame.height {
			
			let location = abs(1 - (bottom / self.frame.height))
			
			self.layer.mask = self.visibilityMaskWithLocation(location, reverse: true)
			
		} else {
			
			self.layer.mask = nil
		}
		
		self.layer.masksToBounds = true
	}
	
	
	func visibilityMaskWithLocation(_ location: CGFloat, reverse: Bool = false) -> CAGradientLayer {
		
		let mask = CAGradientLayer()
		
		mask.frame = self.bounds
		
		if reverse {
			
			mask.colors = [ UIColor(white: 1, alpha: 1).cgColor, UIColor(white: 1, alpha: 0).cgColor ]
		
		} else {
		
			mask.colors = [ UIColor(white: 1, alpha: 0).cgColor, UIColor(white: 1, alpha: 1).cgColor ]
		}
		
		mask.locations = [ NSNumber(value: location.native), NSNumber(value: location.native) ]
		
		return mask
	}
}


extension UITableViewHeaderFooterView {
	
	func setBackgroundTransparent(labelColor: UIColor) {
		
		self.contentView.backgroundColor = UIColor.clear
		
		self.backgroundView?.backgroundColor = UIColor.clear
		
		self.textLabel?.textColor = labelColor
	}
}
